### DISCUSSION TITLE
Welcome to SIP Discussions!

### DISCUSSION THREAD

**Discussion by Dan-in-CA** [Dan-in-CA/SIP#252] [Ideas]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Kenfg** [comment]
Hi
Can I change the hostname of the Raspberry Pi and if so how Please?
I see it as a security issue.
Thanks
Ken

**Reply by Dan-in-CA** [reply]
Changing the host name is a Raspberry pi OS setting rather than a SIP setting.
There a number of web sites with instructions for changing it such as:
https://thepihut.com/blogs/raspberry-pi-tutorials/19668676-renaming-your-raspberry-pi-the-hostname

Dan

**Comment by Kenfg** [comment]
Hi Dan
Thanks for the response.
I have used SIP for many years (with 24 stations) but recently rebuilt a
new sd micro card to upgrade from Wheezy to Buster.
When I installed SIP I had already renamed the pi. The install failed with
a message that it could not find "raspberrypi". I changed the hostname back
to "raspberrypi" and the install ran fine. I have not tried to the change
the hostname since the install.
If it is safe I will change it again, but it would be nice if it installed
on a Pi with a modified hostname?
Thanks in anticipation,
Ken Downsborough
Peth ,
Western Australia.
Mob:      0417 92 888 6
Email:    ken@downsbo.info

**Reply by Dan-in-CA** [reply]
That is strange because SIP can be installed on Beagle Bone Black as well as Raspberry Pi. The installation is simply copying files from GitHub.
Changing the hostname after the install should not be a problem. I will look into it.

The current version of SIP can run under both Python2 and Python3. It is now recommended to use Python3 as the older version is no longer being supported by the Python developers.
I need to update the instructions for setting up sip for automatic start to use Python3. Basically you just substitute sip3.service (or sip3) instead of sip.service (or sip) in the instructions.

That also requires installing Python3 rpi-gpio for Raspbian Buster:
`sudo apt install python-rpi.gpio python3-rpi.gpio`

If python3-rpi.gpio is not installed SIP will appear to run normally but stations will not be activated.

Dan
